File: go-fiber-bun/internal/database/models/user_notes.go
package models

import (
	"time"

	"github.com/uptrace/bun"
)

type UserNotes struct {
	bun.BaseModel `bun:"user_notes"`

	Id      int    `bun:",pk,autoincrement" json:"id"`
	UserId  int    `json:"userId"`
	Title   string `json:"title"`
	Content string `json:"content"`

	CreatedAt time.Time  `bun:",nullzero,notnull,default:current_timestamp" json:"createdAt"`
	UpdatedAt time.Time  `bun:",nullzero,notnull,default:current_timestamp" json:"updatedAt"`
	DeletedAt *time.Time `bun:",soft_delete,nullzero" json:"deletedAt,omitempty"`
}

File: dart-alfred-drift/bin/database/models/user_notes.dart
import 'package:drift/drift.dart';
import 'package:drift_postgres/drift_postgres.dart';

import '../../types/json.dart';
import '../database.dart';
import 'users.dart';

class UserNotes extends Table {
  IntColumn get id => integer().autoIncrement()();

  IntColumn get userId => integer().references(Users, #id)();

  TextColumn get title => text().withLength(max: 100).unique()();

  TextColumn get content => text().unique()();

  TimestampColumn get createdAt => customType(PgTypes.timestampNoTimezone)();

  TimestampColumn get updatedAt => customType(PgTypes.timestampNoTimezone)();

  TimestampColumn get deletedAt => customType(PgTypes.timestampNoTimezone).nullable()();
}

Json userNoteToJson(UserNote note) => {
      "id": note.id,
      "userId": note.userId,
      "title": note.title,
      "content": note.content,
      "createdAt": note.createdAt.dateTime.toIso8601String(),
      "updatedAt": note.updatedAt.dateTime.toIso8601String(),
      if (note.deletedAt case final deletedAt?) "deletedAt": deletedAt.dateTime.toIso8601String(),
    };
